src/test/java/org/jenkinsci/plugins/workflow/steps/durable_task/PowerShellStepTest.java

```diff
@@ -65,6 +65,13 @@
         p.setDefinition(new CpsFlowDefinition("node {powershell 'throw \"bogus error\"'}", true));
         j.assertLogContains("bogus error", j.assertBuildStatus(Result.FAILURE, p.scheduleBuild2(0)));
     }
+
+    // Test that a powershell step that fails indeed causes the underlying build to fail with stopOnFailure: true
+    @Test public void testStopOnFailure() throws Exception {
+        WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "stopfailure");
+        p.setDefinition(new CpsFlowDefinition("node {powershell( script: 'throw \"bogus error\"', stopOnFailure: true)}", true));
```

Comment:
Maybe make the job do more after the error step (like print a line) and verify that the subsequent actions do not appear